test/prompt/invocation_layer/test_hugging_face.py

```diff
@@ -522,6 +522,40 @@ def test_generation_kwargs_from_invoke():
     mock_call.assert_called_with(the_question, {}, {"do_sample": True, "top_p": 0.9, "max_length": 100}, {})
 
 
+@pytest.mark.unit
+def test_max_length_from_invoke(mock_auto_tokenizer, mock_pipeline, mock_get_task):
+    """
+    Test that max_length passed to invoke are passed to the underlying HF model
+    """
+    the_question = "What does 42 mean?"
+    # test that generation_kwargs are passed to the underlying HF model
+    layer = HFLocalInvocationLayer()
+    layer.invoke(prompt=the_question, generation_kwargs={"max_length": 200})
+    invocation_found = False
+    for call in mock_pipeline.mock_calls:
+        # find the call to pipeline invocation, and check that the kwargs are correct
+        if the_question in call.args:
+            found_kwargs = call.kwargs == {"max_length": 200}
+            if found_kwargs:
+                invocation_found = True
+                break
+
+    assert invocation_found
```

Comment:
Let's simplify this:
```suggestion
    # find the call to pipeline invocation, and check that the kwargs are correct
    assert any((call.kwargs == {"max_length": 100}) and (query in call.args) for call in mock_pipeline.mock_calls)
```